### DISCUSSION TITLE
getting cursor on connection

### DISCUSSION THREAD

**Discussion by suraj-ora-2020** [sqlalchemy/sqlalchemy#9920] [Usage Questions]
Hi ,

I am using following code snippet which is using sqlalchemy and the connection pool is created using oracledb pool. After creating the connection which i am trying tp grab the cursor object it is raising error `AttributeError: 'Connection' object has no attribute 'cursor'`

The code goes as

```
from sqlalchemy import create_engine,text
from sqlalchemy.pool import NullPool
import oracledb

pool = oracledb.create_pool(user='xx', password='xx',
                            dsn='xx:1521/xx',
                            min=1, max=3, increment=1)

def creator():
    return pool.acquire(cclass="MYAPP1",purity=oracledb.PURITY_SELF)

engine = create_engine("oracle+oracledb://", creator=creator, poolclass=NullPool)

conn =  engine.connect()
cur = conn.cursor()
sql = "select current_timestamp from dual"
result = cur.execute(sql)
conn.close()

```

**Comment by CaselIT** [comment]
Hi

A sqlalchemy connection does not have a cursor method.
You can look at this part of the docs: https://docs.sqlalchemy.org/en/20/core/connections.html#working-with-the-dbapi-cursor-directly
